src/OrchardCore/OrchardCore.Contents.TagHelpers/ContentLinkTagHelper.cs

```diff
@@ -82,6 +83,27 @@ public override async Task ProcessAsync(TagHelperContext tagHelperContext, TagHe
             if (DisplayFor != null)
             {
                 contentItem = DisplayFor;
+                var previewAspect = await _contentManager.PopulateAspectAsync<PreviewAspect>(contentItem);
+
+                if (!string.IsNullOrEmpty(previewAspect.PreviewUrl))
+                {
+                    var previewUrl = previewAspect.PreviewUrl;
+                    if (!previewUrl.StartsWith("~/", StringComparison.OrdinalIgnoreCase))
```

Comment:
I think this would be useful as an extension method on `PreviewAspect` - it becomes obvious that it may need virtualizing.

`NormalizePath`

Reply:
I agree!  Same logic I am also using for form's action in my contents

Reply:
> If you wish, we can rename `PreviewAspect` and it's `PreviewUrl` property used here as It's not just PreviewUrl rather it's actually `DisplayUrl` for decoupled scenarios.

Let's not rename it at this point, but keep that as an idea / possibility. I suspect @sebastienros may have some thoughts Tuesday.

Lets get `NormalizePath` in and merge it for right now